neuralop/utils.py

```diff
@@ -91,3 +91,47 @@ def get_wandb_api_key(api_key_file='../config/wandb_api_key.txt'):
         with open(api_key_file, 'r') as f:
             key = f.read()
         return key.strip()
+
+# Define the function to compute the spectrum
+def spectrum2(u, s):
+    """This function computes the spectrum of a 2D signal using the Fast Fourier Transform (FFT).
+
+    Paramaters
+    ----------
+    u: a tensor of shape (T * s * s) 
+        A 2D signal represented as a 1D tensor with shape (T * s * s), where T is the number of time steps and s is the spatial size of the signal. T can be any number of channels that we reshape into and s * s is the spatial resolution.
+    s: an integer
+        The spatial size of the signal.
+        
+    Returns
+    --------
+    spectrum: a numpy array 
+        A 1D numpy array of shape (s,) representing the computed spectrum.
+    """
+    T = u.shape[0]
+    u = u.reshape(T, s, s)
+    # u = torch.rfft(u, 2, normalized=False, onesided=False) - depending on your choice of normalization and such
+    u = torch.fft.fft2(u)
+
+    # 2d wavenumbers following Pytorch fft convention
+    k_max = s // 2
+    wavenumers = torch.cat((torch.arange(start=0, end=k_max, step=1), \
```

Comment:
Can we do everything in pytorch instead of going back and forth pytorch <-> numpy?

Reply:
Sure, that is a good suggestion.